envoy/stream_info/stream_info.h

```diff
@@ -602,6 +620,21 @@ class StreamInfo {
    * was never attempted upstream.
    */
   virtual absl::optional<uint32_t> attemptCount() const PURE;
+
+  virtual std::shared_ptr<BytesMeterer> getUpstreamBytesMeterer() const PURE;
+
+  virtual std::shared_ptr<BytesMeterer> getDownstreamBytesMeterer() const PURE;
```

Comment:
@yanavlasov One drawback of this as the interface is that every time we're getting the meterer to update a count we do an atomic inc and dec on the sharedptr ref count.

Reply:
It makes sense to return a reference.